runtime/slice.go

```diff
@@ -156,6 +156,58 @@ func initSliceType(map[string]*Object) {
 	SliceType.slots.Repr = &unaryOpSlot{sliceRepr}
 }
 
+func sliceIndex(f *Frame, o *Object) (i int64, raised *BaseException) {
```

Comment:
Looks like Python only check as Subclass of Int and the actual PyIntObject store the value of Int in long, and cast it with [PyInt_AS_LONG](https://github.com/python/cpython/blob/2.7/Python/ceval.c#L4694).
```
typedef struct {
    PyObject_HEAD
    long ob_ival;
} PyIntObject;
```